zerver/views/registration.py

```diff
@@ -418,6 +419,14 @@ def accounts_register(request: HttpRequest) -> HttpResponse:
             )
             send_initial_realm_messages(realm)
 
+            realm_creation_audit_log = (
+                RealmAuditLog.objects.filter(event_type=RealmAuditLog.REALM_CREATED, realm=realm)
+                .order_by("id")
+                .last()
+            )
+            realm_creation_audit_log.acting_user = user_profile
+            realm_creation_audit_log.save(update_fields=["acting_user"])
```

Comment:
This should be in `do_create_user`, if `realm_creation=True`; in additional to being a cleaner abstraction boundary, it also fixes a bug (that `zproject/backends.py` can be used to create a realm with the LDAP auth backend not using this code path).